src/index.js

```diff
@@ -17,62 +22,107 @@ pressEnter.addEventListener('mouseout', () => {
   hoverText.style.display = 'none';
 });
 
-// Local Storage
-class ToDo {
-  constructor() {
-    this.taskList = JSON.parse(localStorage.getItem('storage-task')) || [];
-  }
-
-  addTask(description) {
-    const updateTask = [
-      ...this.taskList,
-      { description },
-    ];
-    this.updateStorage(updateTask);
-  }
-
-  getFromStorage() {
-    return this.taskList;
-  }
-
-  updateStorage(data) {
-    localStorage.setItem('storage-task', JSON.stringify(data));
-    this.taskList = data;
-  }
-}
-
-const tasks = new ToDo();
-let taskListArray = tasks.getFromStorage();
-
-const showTask = () => {
-  showToDo.innerHTML = '';
-  taskListArray.forEach((element) => showToDo.insertAdjacentHTML('beforeend',
-    `<ul>
-    <li><input type="checkbox" class="checkbox" ${element.completed ? 'checked' : ''}></li>
-    <li>${element.description}</li>
-    <li id="verticle-dot" class="verticle-dot"><i class="fas fa-ellipsis-v"></i></li>
-  </ul>`));
+const showToDo = () => {
+  const tasks = getFromStorage();
+  const tasksList = document.getElementById('tasks-list');
+  tasksList.innerHTML = '';
+  tasks.forEach((todo, index) => {
+    const li = document.createElement('li');
+    li.className = 'new-task';
+
+    const todoItemLeft = document.createElement('div');
+    todoItemLeft.className = 'checkbox-and-task-description';
+
+    const input = document.createElement('input');
+    input.className = 'checkbox';
+    input.type = 'checkbox';
+    if (todo.completed) {
+      input.setAttribute('checked', '');
+    }
+
+    input.onchange = (e) => {
+      if (e.target.checked) {
+        tasks[index].completed = true;
+        e.target.parentNode.children[1].classList.add('strike-through');
+      } else {
+        tasks[index].completed = false;
+        e.target.parentNode.children[1].classList.remove('strike-through');
+      }
+      saveToStorage(tasks);
+    };
+
+    todoItemLeft.appendChild(input);
+
+    const taskDescription = document.createElement('p');
+    taskDescription.classList.add('show');
+    if (todo.completed) {
+      taskDescription.classList.add('strike-through');
+    } else {
+      taskDescription.classList.remove('strike-through');
+    }
+    taskDescription.innerText = todo.description;
+    todoItemLeft.appendChild(taskDescription);
+
+    const editInput = document.createElement('input');
+    editInput.className = 'hide';
+    editInput.type = 'text';
+    editInput.value = todo.description;
+    editInput.addEventListener('focusout', (e) => {
+      li.classList.toggle('edit-bg');
+      updateToDo(tasks, index, e.target.value);
+      showToDo();
+    });
+    todoItemLeft.appendChild(editInput);
+
+    li.appendChild(todoItemLeft);
+
+    const deleteIcon = document.createElement('span');
+    deleteIcon.className = 'hide';
+    deleteIcon.innerHTML = 'delete';
+    deleteIcon.addEventListener('click', () => {
+      deleteToDo(tasks, index);
+      showToDo();
+    });
+    li.appendChild(deleteIcon);
```

Comment:
- [ ] Wonderful job so far. However, there's a little issue with your `deleteTodo`. It is not working at this moment. Kindly look into it and fix. Take a look at the current behaviour of your deleteTodo function.

[app-behave.webm](https://user-images.githubusercontent.com/26597432/220980272-12341b52-f820-4aff-b1f8-c2ecc5c196d6.webm)